### PR TITLE
Move to using our fork of merk

### LINKED ISSUE #89: Move from `fmerk` to `merk`
Goal: makes our life way easier for snapshots recovery, since `fmerk` does not implement the feature at all.

This requires nightly compiler in the repo.

### PR DESCRIPTION
**Pull request by hansl**
This replaces fmerk for our own fork. `merk` has a few advantages and is still heavily maintained.

PR liftedinit/merk#4 should be merged before this one, and the rev updated.

Fixes #89

### CODE REVIEW

File: src/many-ledger/src/lib.rs [hunk 1]
```diff
@@ -17,20 +17,17 @@ pub fn verify_proof(
     symbols: &[Symbol],
     expected_hash: &[u8; 32],
 ) -> Result<BTreeMap<Symbol, TokenAmount>, String> {
-    let keys: Vec<Vec<u8>> = symbols
-        .iter()
-        .map(|s| key_for_account_balance(identity, s))
-        .collect();
-    let values =
-        fmerk::verify_proof(bytes, keys.as_slice(), *expected_hash).map_err(|e| e.to_string())?;
+    let values = merk::verify(bytes, *expected_hash).map_err(|e| e.to_string())?;
```
Comment (fmorency) [inline src/many-ledger/src/lib.rs:20]: Can we test this method?

File: staging/ledger_state.json [hunk 2]
```diff
@@ -45,5 +45,5 @@
       ]
     }
   ],
-  "hash": "ca17a2c678d52849228b7f44f31ffdc4cdf1111b718d97043d7b313793a53a64"
+  "hash": "112598f5158d0a8f95a89dfd3000899ad617999a8445a65032ca81dbc15cf745"
```
Comment (fmorency) [inline staging/ledger_state.json:48]: Why did the hash change?

**Review by fmorency — APPROVED** [review]
Approving assuming comments are addressed, merk rev updated and CI fixed.

File: src/many-ledger/src/lib.rs (hunk 1 above)
Comment (hansl) [reply src/many-ledger/src/lib.rs:20]: I should remove this method, rather.

File: staging/ledger_state.json (hunk 2 above)
Comment (hansl) [reply staging/ledger_state.json:48]: Different hashing technique between merk and fmerk?  I'm not 100% sure.

**Review by fmorency — APPROVED** [review]

**Pull request merged**